src/algorithm/hnswlib/hnswalg.cpp

```diff
@@ -159,7 +159,7 @@ HierarchicalNSW::normalizeVector(const void*& data_point,
 
 float
 HierarchicalNSW::getDistanceByLabel(LabelType label, const void* data_point) {
-    std::unique_lock<std::mutex> lock_table(label_lookup_lock_);
+    std::shared_lock lock_table(label_lookup_lock_);
```

Comment:
L172 have thread-safe issue ?

Reply:
I have placed the writing part in the addPoint function under the protection of the `label_lookup_lock_`